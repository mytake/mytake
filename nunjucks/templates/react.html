{% extends "base.html" %}

{% block endbody %}
  <!-- Dependencies -->
  <script src="https://unpkg.com/react@15/dist/react.js"></script>
  <script src="https://unpkg.com/react-dom@15/dist/react-dom.js"></script>

  <!-- Main -->
  <script src="./bundle-0.0.3.js"></script>
{% endblock %}
