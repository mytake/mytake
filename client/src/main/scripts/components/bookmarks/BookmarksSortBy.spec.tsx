/*
 * MyTake.org website and tooling.
 * Copyright (C) 2020 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
import * as React from "react";
import * as renderer from "react-test-renderer";
import { BookmarksMode } from "./bookmarks";
import BookmarksSortBy from "./BookmarksSortBy";

test("BookmarksSortBy Date Bookmarked", () => {
  const tree = renderer
    .create(
      <BookmarksSortBy
        onChange={jest.fn()}
        selectedOption={BookmarksMode.DateBookmarked}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("BookmarksSortBy Date Happened", () => {
  const tree = renderer
    .create(
      <BookmarksSortBy
        onChange={jest.fn()}
        selectedOption={BookmarksMode.DateHappened}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
