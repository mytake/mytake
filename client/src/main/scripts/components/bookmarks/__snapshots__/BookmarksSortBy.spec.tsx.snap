// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarksSortBy Date Bookmarked 1`] = `
<div
  className="sortby"
>
  <label
    htmlFor="bookmarks-sortby"
  >
    Sort by:
  </label>
  <select
    id="bookmarks-sortby"
    name="sortby"
    onChange={[Function]}
    value="DateBookmarked"
  >
    <option
      value="DateBookmarked"
    >
      Date bookmarked
    </option>
    <option
      value="DateHappened"
    >
      Date happened
    </option>
  </select>
</div>
`;

exports[`BookmarksSortBy Date Happened 1`] = `
<div
  className="sortby"
>
  <label
    htmlFor="bookmarks-sortby"
  >
    Sort by:
  </label>
  <select
    id="bookmarks-sortby"
    name="sortby"
    onChange={[Function]}
    value="DateHappened"
  >
    <option
      value="DateBookmarked"
    >
      Date bookmarked
    </option>
    <option
      value="DateHappened"
    >
      Date happened
    </option>
  </select>
</div>
`;
